### ISSUE TITLE
Auto-generate datasets

### ISSUE THREAD

**Issue by jakevdp** [ellisonbg/altair#112]
I noticed that there are some new datasets available for vega-lite. We should try to auto-generate them from the vega-datasets repo. I would probably generate a ``datasets.json`` file, put it in the package, and read it into the ``_datasets`` variable in the current ``datasets.py``.

A couple other dataset-related thoughts:

- maybe use ``lru_cache`` to cache datasets in memory so they're not downloaded twice within a session
- maybe auto-generate dataset functions like ``load_cars()`` so that they can be tab-completed, with description in the doc-string

**Comment by ellisonbg** [comment]
The other issue is that for both the datasets and examples, we probably
want to pin to the last stable release of vega-lite. Basically, the
examples/datasets we track should match the spec version we are targeting.
I thought I hardcoded the URL in the dataset function to the last vega-lite
release. Thoughts?

On Mon, May 23, 2016 at 7:12 PM, Jake Vanderplas <notifications@github.com>
wrote:

> I noticed that there are some new datasets available for vega-lite. We
> should try to auto-generate them from the vega-datasets repo. I would
> probably generate a datasets.json file, put it in the package, and read
> it into the _datasets variable in the current datasets.py.
>
> A couple other dataset-related thoughts:
>
>    - maybe use lru_cache to cache datasets in memory so they're not
>    downloaded twice within a session
>    - maybe auto-generate dataset functions like load_cars() so that they
>    can be tab-completed, with description in the doc-string
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/ellisonbg/altair/issues/112>
>



-- 
Brian E. Granger
Associate Professor of Physics and Data Science
Cal Poly State University, San Luis Obispo
@ellisonbg on Twitter and GitHub
bgranger@calpoly.edu and ellisonbg@gmail.com

**Comment by jakevdp** [comment]
Yes, I think version consistency is very important. I already ran into that when I tried to auto-generate examples from vega-lite master: a couple of them used opacity as an encoding option, which is invalid under the current spec.

For sustainability, I'd propose to create scripts to download the spec, the datasets, and the examples given a vega release number. Then we can run something along the lines of ``sync_with_vega.py --version=1.1`` and have everything (*fingers crossed*) happen automatically.
